src/plugin/plugin_curl.rs

```diff
@@ -404,8 +404,13 @@ impl CurlPlugin {
     }
 
     fn inject_sw_header(request_id: Option<i64>, ch: ZVal, info: &CurlInfo) -> crate::Result<()> {
-        let sw_header = RequestContext::try_with_global_ctx(request_id, |ctx| {
-            Ok(encode_propagation(ctx, info.url.path(), &info.peer))
+        let sw_header = RequestContext::try_with_global(request_id, |req_ctx| {
+            let span_object = req_ctx.entry_span.span_object();
```

Comment:
Is there any case the entry_span is null? Usually, in Java practice. we would pick the first entry span, if it can't be found, we fall back to the first span. This is usually happens this trace starts from a timer/scheduler.

Reply:
Or sometimes, the plugin doesn't cover the entry. So, only the current exit span with some local spans left.

Reply:
Now PHP trace always start from HTTP requests, so the entry span will exist, I think.

Reply:
I just suggest maybe it is better to add a method to retrieve the endpoint(looking for entrySpan's) rather than hard-coding this everywhere.

Reply:
Well, I has switch to use `get_primary_span` method.